app/Navigation/NavigationMetaFields.php

```diff
@@ -0,0 +1,81 @@
+<?php
+
+namespace App\Navigation;
+
+class NavigationMetaFields {
+
+	/**
+	 * NavigationMetaFields constructor.
+	 */
+	public function __construct() {
+		add_action( 'wp_nav_menu_item_custom_fields', [ self::class, 'addCustomFields' ], 10, 2 );
+		add_action( 'wp_update_nav_menu_item', [ self::class, 'updateNavItems' ], 10, 2 );
+	}
+
+	/**
+	 * Mapping of the meta fields to the menu item.
+	 *
+	 * @data-structure: 'FIELD_DESCRIPTION' => [
+	 *    'meta_field' => 'META_FIELD_NAME',
+	 *    'text' 	   => 'TEXT_SHOWN_BESIDE_CHECKBOX'
+	 * ]
+	 *
+	 * @return array[]
+	 */
+	public static function getFields() {
+		return [
+			'counter' => [
+				'meta_field' => '_show-counter',
+				'text'       => __( 'Show counter next to this item', 'owc-formulieren' ),
+			],
+			'spacing' => [
+				'meta_field' => '_spacing-top',
+				'text'       => __( 'Add spacing to the top of this item', 'owc-formulieren' ),
+			],
+		];
+	}
+
+	/**
+	 * Add custom fields to the menu item edit screen form.
+	 *
+	 * @param $item_id
+	 * @param $item
+	 *
+	 * @return void
+	 */
+	public static function addCustomFields( $item_id, $item ) {
```

Comment:
Niet persé fout maar waarom zijn de methods static?

Reply:
Eerlijke antwoord is gewenning/preference van schrijven. Zelf vind ik class::function() mooier staan dan class->function(), in ieder geval op plaatsen waar het eigenlijk vrij weinig uit maakt zoals hier.

Ik probeer echter wel de stijl van het project zo veel mogelijk over te nemen dus als je liever non-static functies hebt dan pas ik dat uiteraard aan! :-)

Reply:
In deze context werkt dit prima alleen strookt het met datgeen wat ik geleerd heb. Wellicht dat deze klasse ooit uitgebreid wordt waardoor je de static methods alsnog moet vervangen. Stukje [leesvoer](https://stackoverflow.com/questions/33705976/when-should-i-use-static-methods) voor de liefhebber.